repos/system_upgrade/common/actors/systemfacts/libraries/systemfacts.py

```diff
@@ -242,3 +258,25 @@ def _get_firewall_status(service_name):
 def get_firmware():
     firmware = 'efi' if os.path.isdir('/sys/firmware/efi') else 'bios'
     return FirmwareFacts(firmware=firmware)
+
+
+@aslist
+def _default_grub_info():
+    for line in run(['cat', '/etc/default/grub'], split=True)['stdout']:
+        name, value = tuple(map(type(line).strip, line.split('=', 1)))
+        yield DefaultGrub(
+            name=name,
+            value=value
+        )
+
+
+def get_default_grub():
+    """ Get a list of GRUB parameters from /etc/default/grub """
+    return DefaultGrubInfo(default_grub_info=_default_grub_info())
+
+
+def get_grubcfg():
+    """ Get '/boot/grub2/grub.cfg' content """
+    with open('/boot/grub2/grub.cfg') as fo:
+        grub_cfg = fo.read()
+    return GrubCfg(grub_cfg=grub_cfg)
```

Comment:
is there a reason to store just the file instead of parsed / processed content? storing the text file and than keep everyone parsing it again and again. that's opposite what we tell people they should do when they create scanners and models. this is not good design and example

Reply:
as well, thinking about these scanners, they do not seem to be valid for all arches. Including that /boot/grub2/grub.cfg is not correct path on intel for EFI if I remember well. (let's check that)

Reply:
aah. sorry. I've exchanged the file with the /etc/default/grub :3 but instead of the file itself, let's share a usable info only. e.g. info about BLS in this case.

Reply:
so checked, the file does not exist on EFI (and we are not sure how to correctly detect it as it could have various names, so the model should be created only on non-efi systems right now.

@smitterl I expect this file does not exist on s390x at all as there is used ZIPL instead of grub, correct?